src/lib/snack-bar/snack-bar.ts

```diff
@@ -39,43 +39,47 @@ export class MdSnackBar {
    * Creates and dispatches a snack bar with a custom component for the content, removing any
    * currently opened snack bars.
    */
-  openFromComponent<T>(component: ComponentType<T>,
-                       config: MdSnackBarConfig): MdSnackBarRef<T> {
+  openFromComponent<T>(component: ComponentType<T>, config?: MdSnackBarConfig): MdSnackBarRef<T> {
+    config = this._applyConfigDefaults(config);
     let overlayRef = this._createOverlay();
     let snackBarContainer = this._attachSnackBarContainer(overlayRef, config);
-    let mdSnackBarRef = this._attachSnackbarContent(component, snackBarContainer, overlayRef);
+    let snackBarRef = this._attachSnackbarContent(component, snackBarContainer, overlayRef);
 
     // When the snackbar is dismissed, clear the reference to it.
-    mdSnackBarRef.afterDismissed().subscribe(() => {
+    snackBarRef.afterDismissed().subscribe(() => {
       this._snackBarRef = null;
     });
 
     // If a snack bar is already in view, dismiss it and enter the new snack bar after exit
     // animation is complete.
     if (this._snackBarRef) {
       this._snackBarRef.afterDismissed().subscribe(() => {
-        mdSnackBarRef.containerInstance.enter();
+        snackBarRef.containerInstance.enter();
       });
       this._snackBarRef.dismiss();
     // If no snack bar is in view, enter the new snack bar.
     } else {
-      mdSnackBarRef.containerInstance.enter();
+      snackBarRef.containerInstance.enter();
     }
     this._live.announce(config.announcementMessage, config.politeness);
-    this._snackBarRef = mdSnackBarRef;
+    this._snackBarRef = snackBarRef;
     return this._snackBarRef;
   }
 
   /**
-   * Creates and dispatches a snack bar.
+   * Opens a snackbar with a message and an optional action.
+   * @param message The message to show in the snackbar.
+   * @param action The label for the snackbar action.
+   * @param config Additional configuration options for the snackbar.
+   * @returns {MdSnackBarRef<SimpleSnackBar>}
    */
-  open(message: string, actionLabel: string,
-       config: MdSnackBarConfig): MdSnackBarRef<SimpleSnackBar> {
+  open(message: string, action = '', config?: MdSnackBarConfig): MdSnackBarRef<SimpleSnackBar> {
```

Comment:
Is there a reason to not just set a default config of {}?

open(message: string, action = '', config: MdSnackBarConfig = {})...